zerver/tests/test_narrow.py

```diff
@@ -541,6 +553,24 @@ def test_get_old_messages_with_narrow_stream_mit_unicode_regex(self):
             stream_id = stream_messages[i].recipient.id
             self.assertEqual(message["recipient_id"], stream_id)
 
+    def test_get_old_messages_with_narrow_stream_mit_unicode_null_regex(self):
+        # type: () -> None
+        self.login("starnine@mit.edu")
+        null_stream_name = u"foo\000-stream"
+        self.subscribe_to_stream("starnine@mit.edu", null_stream_name)
+        self.send_message("starnine@mit.edu", null_stream_name, Recipient.STREAM)
+        narrow = [dict(operator='stream', operand=null_stream_name)]
+        result = self.get_and_check_messages(dict(num_before=2, num_after=2,
+                                                  narrow=ujson.dumps(narrow)))
+
+        messages = get_user_messages(get_user_profile_by_email("starnine@mit.edu"))
+        stream_messages = [msg for msg in messages if msg.recipient.type == Recipient.STREAM]
+        self.assertEqual(len(result["messages"]), 1)
+        for i, message in enumerate(result["messages"]):
```

Comment:
This was copied to match the pattern of the previous test's validation.  I'll get rid of it, but the code doesn't get much less messy :/